src/claimtrie.h

```diff
@@ -255,24 +241,77 @@ class CClaimIndexElement
         READWRITE(claim);
     }
 
+    bool empty() const
+    {
+        return name.empty() && claim.claimId.IsNull();
+    }
+
     std::string name;
     CClaimValue claim;
 };
 
```

Comment:
This block of code (253 - 293) is a bit confusing, it's basically just to enable swap() right? 
Maybe adding some comments about what its trying to achieve would be helpful.

Reply:
From 262 to 284 - yes, it for std::swap to work. Helpers and typedefs are to make a different type when insert to db, so let's see an example
```
typedef Generic<outPointHeightType, queueNameRowHelper> queueNameRowValueType;
typedef Generic<outPointHeightType, supportQueueNameRowHelper> supportQueueNameRowValueType;
```
Helper is used only to make these 2 types different e.g. to stored in different keys in db. I'll commented it.

Reply:
line 253: /// Helpers to separate queue types from each other
line 271: /// Make std::swap to work with custom types
line 288: /// Each type will be stored in database separately

Reply:
looks good